StreamChatClient_v3/WebSocketClient/Engine/URLSessionWebSocketEngine.swift

```diff
@@ -71,21 +70,20 @@ final class URLSessionWebSocketEngine: NSObject, WebSocketEngine, URLSessionData
                   webSocketTask: URLSessionWebSocketTask,
                   didCloseWith closeCode: URLSessionWebSocketTask.CloseCode,
                   reason: Data?) {
-    var error: WebSocketProviderError?
+    var error: WebSocketEngineError?
 
     if let reasonData = reason, let reasonString = String(data: reasonData, encoding: .utf8) {
-      error = WebSocketProviderError(
+      error = WebSocketEngineError(
         reason: reasonString,
-        code: 0,
-        providerType: URLSessionWebSocketEngine.self,
+        code: closeCode.rawValue,
         providerError: nil
       )
     }
 
     disconnect(with: error)
   }
 
-  private func disconnect(with error: WebSocketProviderError?) {
+  private func disconnect(with error: WebSocketEngineError?) {
     isConnected = true
```

Comment:
This is what we fixed in the last release.
```suggestion
    isConnected = false
```